server/src/main/java/org/jboss/as/server/deployment/DeploymentOverlayDeploymentUnitProcessor.java

```diff
@@ -65,6 +66,8 @@
 
     private final ContentRepository contentRepository;
 
+    private static final String WAR_EXTENSION = ".war";
```

Comment:
Only in the case of overlay for resources like `war/some.file` within an ear deployment, the method `isExplodedSubUnitOverlay(DeploymentUnit deploymentUnit, VirtualFile mountPoint, String path)` is called(), the `deploymentUnit` in the parameters is the top level `ear` deployment, the  `org.jboss.as.server.deployment.MountExplodedMarker.isMountExploded(DeploymentUnit deploymentUnit)` will return false eventually, so this does not work.

After more research, I have 2 options:

* I see that the `Attachments.INDEX_RESOURCE_ROOT` will be attached to the war resource root by [EarStructureProcessor](https://github.com/wildfly/wildfly/blob/master/ee/src/main/java/org/jboss/as/ee/structure/EarStructureProcessor.java#L279), so maybe that can be used to check whether some resources in war of ear are overlayed, like:

<pre>
&& resourceRoot.hasAttachment(Attachments.INDEX_RESOURCE_ROOT)){...}
</pre>

>  This option is more tricky, but only changes in wildfly-core can solve the problem.

* Change the `MountExplodedMarker` to be able to set exploded flag for `ResourceRoot` too, and in `EarStructureProcessor`, puts exploded flag to the war resource root, then check the exploded flag in `DeferredDeploymentOverlayDeploymentUnitProcessor`

> This option needs both changes in wildfly-core and wildfly to solve the problem.

If one of above option is OK, which one do you prefer? Thank you.